TAO/orbsvcs/orbsvcs/SSLIOP/SSLIOP_Factory.cpp

```diff
@@ -547,6 +549,15 @@ TAO::SSLIOP::Protocol_Factory::init (int argc, ACE_TCHAR* argv[])
         {
           this->check_host_ = true;
         }
+      else if (ACE_OS::strcasecmp(argv[curarg],
+                                  ACE_TEXT("-SSLEcName")) == 0)
+      {
+          curarg++;
+          if (curarg < argc)
+          {
+              ec_name = static_cast<const char *>(ACE_TEXT_ALWAYS_CHAR(argv[curarg]));
```

Comment:
This cast looks suspicious, it is not used for any other ssl argument

Reply:
It is required to use the copying assignment instead of the owning assignment of String_var.

Reply:
In other places a string_dup is used, why not here, that makes very clear a copy is made

Reply:
Using `const char*` we can interact with String_var/String_Manager and all the normal C++ behaviors for copy/assign/destruct remain in effect (without explicit memory management).  Unfortunately the same isn't true for `char*`.  It would be helpful if `init()` functions that shouldn't be changing the data used a type like `const char* argv[]` so that const could be maintained throughout, but when interfacing with older APIs we may need to add it explicitly.